packer/fck-nat.pkr.hcl

```diff
@@ -96,7 +96,8 @@ build {
 
   provisioner "shell" {
     inline = [
-      "sudo yum --nogpgcheck -y localinstall /tmp/fck-nat-${var.version}-any.rpm"
+      "sudo yum --nogpgcheck -y localinstall /tmp/fck-nat-${var.version}-any.rpm",
+      "sudo yum install amazon-cloudwatch-agent -y"
```

Comment:
awesome stuff @RaJiska  - I'm using your TF module (will do a PR with some changes I made for my use cases)

```suggestion
```

is it possible to add `depends on` in fpm to define the dependency on amazon-cloudwatch-agent?

https://github.com/AndrewGuenther/fck-nat/blob/519bca3746a750ee5e29b4738ab99d0c82efd016/.fpm#L9C1-L9C32

altho, after reading AWS docs
https://docs.aws.amazon.com/systems-manager/latest/userguide/agent-install-al2.html#quick-install-al2

maybe it's not possible, but wouldn't it be better part of the post-install script invoked by the RPM to stay consistent?

https://github.com/AndrewGuenther/fck-nat/blob/519bca3746a750ee5e29b4738ab99d0c82efd016/service/post-install.sh

Reply:
Thanks @vincentgna , happy to know the Terraform module can be helpful to you!

Is there any downside in not using dependency? Also, should any package that is installed for fck-nat be registered as a dependency?
Cloudwatch Agent is one, but some might be added in the future as well, the Amazon SSM Agent for instance.

Reply:
correct, I added the SSM Agent to my `.fpm` and changed your module to associate the SSM iam policy to the NAT instance role. I also added terratest to built and validate the AMI in both HA and non-HA set up from your module (I'm using a monorepo to achieve this).

Reply:
My general philosophy on this has been that anything necessary to core functions should be a direct dependency of the RPM and anything necessary to support optional features will be installed in the AMIs. So I wouldn't add them here.

Also, the base AMI I _believe_ already has the SSM agent installed. If that's not the case, then yes, we should add it here.

Reply:
cool, then I'll resolve my comment re: moving the cloudwatch-agent to the rpm (as it is not a direct dependency of the RPM).

As for the SSM Agent, in the ARM Amzn2 Minimal base AMI I tested yesterday, it was missing and I had to added the SSM Agent. I assume it wasn't in fck-nat perhaps for it to act more like an AWS "appliance" - but I couldn't troubleshoot my NAT without adding EC2 keypair and opening port 22 ingress, so it was a bit annoying (and it's much more convenient to use SSM Agent, perhaps a separate PR?)

Reply:
Gonna quickly walk back my previous comment: I don't believe the SSM agent is necessary for this change to work. Given that, I _wouldn't_ install it here.

@vincentgna I agree with you that the SSM agent is helpful, but the goal is to keep the base AMIs as slim as possible. I just double checked and in our deployment we're actually installing SSM in our user data. Is there a reason that wouldn't work for you?

Reply:
Yes, I can confirm SSM Agent is not installed by default on the image. And actually I believe it'd be better to disable entirely SSH in the AMI build, and leave it up to the user to re-enable if they wish to maintain a an alternative to SSM.

Reply:
I guess a separate PR/Issue to track the SSH set up and a caveat / docs for end users on how to troubleshoot would help, I agree with minimal set up.

seems this conv is a PR review, I can't flag it as "resolved"

Reply:
@RaJiska I 100% agree on disabling SSH and I consider not doing so one of my "original sins" in this project, so to speak. However, disabling SSH now would be a backwards incompatible change and I would consider most appropriate to track for a 2.0 release. I'll open an issue for it and create a 2.0 milestone as well.

I'll use the 2.0 milestone not to target a specific release date, but more as a bucket to hold breaking changes and once that list becomes compelling enough we can formalize a release for it.

Reply:
Filed #47

Resolving this thread.